app/Services/CampaignService.php

```diff
@@ -176,4 +177,33 @@ public function getCampaignIdsFromSignups()
 
         return $ids ? $ids : null;
     }
+
+    /**
+     * Appends status counts to a collection of campaigns.
+     *
+     * @return Illuminate\Database\Eloquent\Collection $campaigns
+     */
+    public function getCampaignPostStatusCounts($campaigns)
+    {
+        $ids = array_pluck($campaigns, 'id');
+
+        $campaignsWithSignups = Signup::campaign($ids)
+            ->includePostStatusCounts()
+            ->get()
+            ->groupBy('campaign_id');
```

Comment:
Oh hmmm, I feel like you should be able to do that with the query builder but I'm not particularly morally opposed to raw SQL queries either (especially for a pretty unique thing like this!)

Before going too far down this route I would just double-check that `SUM()` function can use indexes and wouldn't get too slow if we had a bajillion records in the database.